### PR TITLE
feat: View the currently known proving live sectors information

### PR DESCRIPTION
**Pull request by swift-mx**
## Checklist

Before you mark the PR ready for review, please make sure that:
- [x] All commits have a clear commit message.
- [x] The PR title is in the form of of `<PR type>: <area>: <change being made>`
    - example: ` fix: mempool: Introduce a cache for valid signatures`
    - `PR type`: _fix_, _feat_, _INTERFACE BREAKING CHANGE_, _CONSENSUS BREAKING_, _build_, _chore_, _ci_, _docs_,_perf_, _refactor_, _revert_, _style_, _test_
    - `area`: _api_, _chain_, _state_, _vm_, _data transfer_, _market_, _mempool_, _message_, _block production_, _multisig_, _networking_, _paychan_, _proving_, _sealing_, _wallet_, _deps_
- [x] This PR has tests for new functionality or change in behaviour
- [x] If new user-facing features are introduced, clear usage guidelines and / or documentation updates should be included in https://lotus.filecoin.io or [Discussion Tutorials.](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [x] CI is green

### CODE REVIEW

**Review by magik6k — COMMENTED** [review]
Thanks for the PR!

One suggestion which may make it more useful in the future.

We should also switch `proving deadlines` to `LiveSectors`; AllSectors are not the right thing to use there.

File: cmd/lotus-miner/proving.go [hunk 1]
```diff
@@ -561,3 +562,59 @@ It will not send any messages to the chain.`,
 		return nil
 	},
 }
+
+var provingLivesCmd = &cli.Command{
+	Name:  "lives",
```
[the remark this answers is not in the document]
Comment (swift-mx) [reply cmd/lotus-miner/proving.go]: A new commit has been submitted， please review

File: cmd/lotus-miner/proving.go [hunk 2]
```diff
@@ -238,29 +249,51 @@ var provingDeadlinesCmd = &cli.Command{
...
+					_, _ = fmt.Fprintf(tw, "%d\t%d\t%d (%d|%d|%d|%d)\t\n", dlIdx, partIdx, sc, fc, rc, lc, ac)
+				}
 			}
 
 			var cur string
 			if di.Index == uint64(dlIdx) {
 				cur += "\t(current)"
 			}
-			_, _ = fmt.Fprintf(tw, "%d\t%d\t%d (%d)\t%d%s\n", dlIdx, len(partitions), sectors, faults, provenPartitions, cur)
+			_, _ = fmt.Fprintf(tw, "%d\t%d\t%d (%d|%d|%d|%d)\t%d%s\n", dlIdx, len(partitions), sectors, faults, recovering, live, active, provenPartitions, cur)
```
Comment (geoff-vball) [inline cmd/lotus-miner/proving.go:296]: I think that for the detailed view, we shouldn't print this info for the deadline, it makes the printout a little too confusing.

File: cmd/lotus-miner/proving.go [hunk 3]
```diff
@@ -238,29 +249,51 @@ var provingDeadlinesCmd = &cli.Command{
...
+				}
+				recovering += rc
+
+				lc, err := partition.LiveSectors.Count()
+				if err != nil {
+					return err
+				}
+				live += lc
+				if cctx.IsSet("detailed") {
+					_, _ = fmt.Fprintf(tw, "%d\t%d\t%d (%d|%d|%d|%d)\t\n", dlIdx, partIdx, sc, fc, rc, lc, ac)
```
Comment (geoff-vball) [inline cmd/lotus-miner/proving.go:288]: I think we should just create tabbed columns for all the new data we're displaying.

File: cmd/lotus-miner/proving.go [hunk 4]
```diff
@@ -321,12 +354,12 @@ var provingDeadlineInfoCmd = &cli.Command{
...
 		for pIdx, partition := range partitions {
-			sectorCount, err := partition.AllSectors.Count()
+			sectorCount, err := partition.ActiveSectors.Count()
 			if err != nil {
 				return err
 			}
 
-			sectorNumbers, err := partition.AllSectors.All(sectorCount)
+			sectorNumbers, err := partition.ActiveSectors.All(sectorCount)
 			if err != nil {
```
Comment (geoff-vball) [inline cmd/lotus-miner/proving.go:363]: Let's add a flag `all` to this function to give the user the option to get info for `AllSectors`

**Comment by geoff-vball** [comment]
@swift-mx Did you push the wrong commit? None of the changes in the latest commit seem to line up with my comments. Let me know if you have any questions :)

**Comment by swift-mx** [comment]
> @swift-mx Did you push the wrong commit? None of the changes in the latest commit seem to line up with my comments. Let me know if you have any questions :)

The reason for this PR is that some of our sectors have expired. We want to compress the partitions. But I don't know which one should be compressed. So there is a need

File: cmd/lotus-miner/proving.go (hunk 4 above)
Comment (swift-mx) [reply cmd/lotus-miner/proving.go]: maybe AllSectors is useless to the user

**Comment by magik6k** [comment]
This has developed some conflicts (likely after merging https://github.com/filecoin-project/lotus/pull/8721 / https://github.com/filecoin-project/lotus/pull/9109)

Also note that https://github.com/filecoin-project/lotus/pull/9109 has added similar functionality, just to a different command

**Comment by swift-mx** [comment]
> This has developed some conflicts (likely after merging #8721 / #9109)
> 
> Also note that #9109 has added similar functionality, just to a different command

OK, it's just different display from https://github.com/filecoin-project/lotus/pull/9109 , i will close this pr

**Pull request closed without merging**